trl/trainer/grpo_trainer.py

```diff
@@ -450,6 +450,20 @@ def _move_model_to_vllm(self):
         ) as unwrapped_model:
             if is_compiled_module(unwrapped_model):
                 state_dict = unwrapped_model._orig_mod.state_dict()
+            elif isinstance(unwrapped_model, PeftModel):
+                unwrapped_model.merge_adapter()
+                state_dict = unwrapped_model.state_dict()
+                unwrapped_model.unmerge_adapter()
+                state_dict = {
+                    k.removeprefix("base_model.model.").replace(".base_layer", ""): v
+                    for k, v in state_dict.items()
+                    if self.model.prefix not in k
+                }
+                state_dict = {
+                    k.replace("modules_to_save.default.", ""): v
+                    for k, v in state_dict.items()
+                    if "original_module" not in k
+                }
```

Comment:
Good catch to think about the `modules_to_save`. Note here that the adapter name can be different from `"default"`. You could get the adapter name from `model.active_adapters`, which is a list of all active adapters. I assume in this context, there can only ever be one (raise an error when more?), so taking the first item should work.

Reply:
thanks for reviewing @BenjaminBossan! I'll make it multi-adaptater-proof, probably simpler.